ruff.toml

```diff
@@ -4,33 +4,40 @@
 # This is the exact version of Ruff we use.
 required-version = "0.0.272"
 
-# Enable pycodestyle (`E`) and Pyflakes (`F`) codes by default.
-# D104 - missing docstring in public package
-# D2 - docstring formatting
-# D3 - docstrings quotes and escape types
-# D405-D409 - NumPy docstrings
-# D419 - no empty docstrings
-# TID252 - no relative imports
-# SIM103 - simplify code: needless-bool
-# SIM2** - simplify code: complex equality statements
-# SIM3** - simplify code: yoda-conditions
-# SIM4** - simplify code: if-else-blocks
-# SIM9** - simplify code: dict-get-with-none-default
-select = ["E", "F", "D104", "D2", "D3", "D405", "D406", "D407", "D408", "D409", "D41", "SIM103", "SIM2", "SIM3", "SIM4", "SIM9", "TID"]
+# Enable pycodestyle (E) and Pyflakes (F) codes by default.
+# D - NumPy docstring rules
+# N801 - Class name should use CapWords convention
+# SIM - code simplification rules
+# TID - tidy imports
+select = ["E", "F", "D", "N801", "SIM", "TID"]
 
+# Ruff rules we ignore (for now) because they are not 100% automatable
+#
+# D100 - Missing docstring in public module
+# D101 - Missing docstring in public class
+# D102 - Missing docstring in public method
+# D103 - Missing docstring in public function
+# D106 - Missing docstring in public nested class
+# D401 - First line of docstring should be in imperative mood
+# D404 - First word of the docstring should not be "This"
+# SIM102 - Use a single if statement instead of nested if statements
+# SIM105 - Use contextlib.suppress({exception}) instead of try-except-pass
+# SIM108 - Use ternary operator {contents} instead of if-else-block
+# SIM114 - Combine if branches using logical or operator
+# SIM115 - Use context handler for opening files
+# SIM117 - Use a single with statement with multiple contexts instead of nested with statements
+
+# Ruff rules we ignore because we don't want them
+#
 # D105 - we don't need to document well-known magic methods
 # D205 - 1 blank line required between summary line and description
 # E501 - line length, because we use Black for that
 # E731 - we can use lambdas however we want
-# N802 - lowercase function naming, we use camel case
-# N803 - lowercase argument naming, we use camel case
-# N806 - lowercase variable naming, we use camel case
-# N815 - mixed case in class header, we use camel case
-# N816 - mixed case in global, scripts can use mixed case
-# N999 - lowercase variable naming, we use camel case
+# N - PEP8 naming conventions (we use camelCase; too late to change)
```

Comment:
N isn't included in the `ignore` list below. We do want to ignore some of the N rules, but not all correct?

Reply:
removed